packages/opentelemetry-basic-tracer/src/BasicTracer.ts

```diff
@@ -60,11 +62,28 @@ export class BasicTracer implements types.Tracer {
       return NOOP_SPAN;
     }
 
-    const spanOptions = Object.assign({}, options, {
-      parent: parentSpanContext,
-    });
-    const span = new Span(this, name, spanOptions);
+    let traceId;
+    let parentSpanId;
+    let traceState;
+    if (!parentSpanContext || !isValid(parentSpanContext)) {
+      // New root span.
+      traceId = randomTraceId();
+    } else {
+      // New child span.
+      traceId = parentSpanContext.traceId;
+      parentSpanId = parentSpanContext.spanId;
+      traceState = parentSpanContext.traceState;
+    }
 
+    const span = new Span(
+      this,
+      name,
+      traceId,
+      options.kind || types.SpanKind.INTERNAL,
```

Comment:
rather than deconstructing the context, can we create and pass in the context the span will use? e.g.

```ts
let context = new SpanContext()
if (!parentContext) {
  context.traceId = randomTraceID();
} else {
  context.traceId = parentContext.traceId
  context.parentSpanId = parentContext.spanId;
  // tracestate 
}
```

and in the span constructor
```ts
if (context.spanId == null) {
  context.spanId = randomSpanID()
}
this._spanContext = context
```

Reply:
Could you briefly outline which problem this solves?

Reply:
@bg451 What do you think about this?
```
let parentSpanContext = this._getParentSpanContext(options.parent);
if (!parentSpanContext || !isValid(parentSpanContext)) {
  // New root span.
  parentSpanContext = { traceId: randomTraceId(), spanId: '' };
}

const span = new Span(this, name, parentSpanContext, 
options.kind || types.SpanKind.INTERNAL,options.startTime);
```

Reply:
Sweet yeah that looks great, thanks!

I saw question in an email about what problem this solves, the answer is that I mostly want to reduce argument bloat. Additionally if any new fields get added to the span context object, it's another place that needs to be updated.

Reply:
@bg451 - yes that was me, thank you. It became obvious Mayurs solution so I deleted my remark.